File: src/main/java/bootcamp/Snap.java
package bootcamp;

import java.util.Scanner;

class Snap {
    private int player1Score;
    private int player2Score;

    public static void main(String[] args) {
        Snap snap = new Snap();
        AnimalDeck deck = new AnimalDeck();
        deck.shuffle();
        AnimalCard previousCard = null;
        Scanner scanner = new Scanner(System.in);

        while (deck.getCards().length > 0) {
            AnimalCard currentCard = deck.deal();
            System.out.println(currentCard);
            String input = scanner.nextLine();
            if (input.length() > 0 && input.charAt(0) == 'a') {
                if (currentCard.snap(previousCard)) {
                    System.out.println("SNAP! score Player 1");
                    snap.player1Score++;
                } else {
                    System.out.println("WRONG! deducting score from Player 1");
                    snap.player1Score--;
                }
            } else if (input.length() > 0 && input.charAt(0) == 'l') {
                if (currentCard.snap(previousCard)) {
                    System.out.println("SNAP! scorePlayer 2");
                    snap.player2Score++;
                } else {
                    System.out.println("WRONG! deducting score from Player 2");
                    snap.player2Score--;
                }
            }
            previousCard = currentCard;
        }

        if (snap.player1Score == snap.player2Score) {
            System.out.println("Draw.");
        } else if (snap.player1Score > snap.player2Score) {
            System.out.println("Player 1 wins!");
        } else System.out.println("Player 2 wins!");

        System.out.println("Scores: " + snap.player1Score + " vs " + snap.player2Score);
    }
}

File: exercises/dotnet/McrDigital.Bootcamp1.Cards/Snap.cs
namespace McrDigital.Bootcamp1.Cards
{
    using System;

    public class Snap
    {
        static void Main(string[] args)
        {
            var snap = new Snap(new PlayingCardDeck());
            snap.Play();
        }

        private int _player1Score;
        private int _player2Score;
        private readonly Deck _deck;

        public Snap(Deck deck)
        {
            _deck = deck;
            _deck.Shuffle();
        }

        public void Play()
        {
            Card previousCard = null;
            while (_deck.GetCards().Length > 0)
            {
                var currentCard = _deck.Deal();
                Console.WriteLine(currentCard);

                var input = Console.ReadLine();
                if (input.Length > 0 && input[0] == 'a')
                {
                    if (currentCard.Snap(previousCard))
                    {
                        Console.WriteLine("SNAP! score Player 1");
                        _player1Score++;
                    }
                    else
                    {
                        Console.WriteLine("WRONG! deducting score from Player 1");
                        _player1Score--;
                    }
                }
                else if (input.Length > 0 && input[0] == 'l')
                {
                    if (currentCard.Snap(previousCard))
                    {
                        Console.WriteLine("SNAP! score Player 2");
                        _player2Score++;
                    }
                    else
                    {
                        Console.WriteLine("WRONG! deducting score from Player 2");
                        _player2Score--;
                    }
                }

                previousCard = currentCard;
            }

            if (_player1Score == _player2Score)
            {
                Console.WriteLine("Draw.");
            }
            else if (_player1Score > _player2Score)
            {
                Console.WriteLine("Player 1 wins!");
            }
            else
            {
                Console.WriteLine("Player 2 wins!");
            }

            Console.WriteLine($"Scores: {_player1Score} vs {_player2Score}");

            //Just wait for input before closing console
            Console.ReadLine();
        }
    }
}